packages/angular_devkit/build_optimizer/src/transforms/wrap-enums.ts

```diff
@@ -0,0 +1,308 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import * as ts from 'typescript';
+import { collectDeepNodes, drilldownNodes } from '../helpers/ast-utils';
+
+
+export const wrapEnumsRegexes = [
+  // tslint:disable:max-line-length
```

Comment:
Seriously consider making those regexes as parser functions. There are many backreferences and those are _really_ slow. I wouldn't be surprised if this is O(n^4), considering you're going to 4 backrefs. Considering you run this on really long strings I think you'd see some improvements. We can chat about it later.